app/components/report-row/component.js

```diff
@@ -26,9 +27,9 @@ export default class ReportRowComponent extends Component {
    * @method save
    * @public
    */
-  @action
-  save(changeset) {
-    this.args.onSave(changeset);
+  @dropTask
+  *save(changeset) {
+    yield this.args.onSave(changeset);
```

Comment:
```suggestion
    if(this.args.onSave) {
      yield this.args.onSave(changeset);
    }
```

Since we switched to a task we now use the `perform` helper, but this won't check if the passed argument is a function or not. (...well technically it does, but it will only yield an assertion error at in the dev env and an error on prod) Earlier we had the `optional` helper for it. So now you have to check by your own if this is a valid function or at least if it's existing.